src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2244,3 +2247,158 @@ def lists_match(l1, l2):
+
+    return client.verify_ip_flow(
+        watcher_rg, watcher_name,
+        VerificationIPFlowParameters(
+            vm, direction, protocol, local_port, remote_port,
+            local_ip_address, remote_ip_address, nic))
+
+
+def show_nw_next_hop(client, resource_group_name, vm, watcher_rg, watcher_name,
+                     source_ip, dest_ip, nic=None, location=None):
+    from azure.mgmt.network.models import NextHopParameters
+
+    if not is_valid_resource_id(vm):
+        vm = resource_id(
+            subscription=get_subscription_id(), resource_group=resource_group_name,
+            namespace='Microsoft.Compute', type='virtualMachines', name=vm)
+
+    if nic and not is_valid_resource_id(nic):
+        nic = resource_id(
+            subscription=get_subscription_id(), resource_group=resource_group_name,
+            namespace='Microsoft.Network', type='networkInterfaces', name=nic)
+
+    return client.get_next_hop(
+        watcher_rg, watcher_name, NextHopParameters(vm, source_ip, dest_ip, nic))
+
+
+def show_nw_security_view(client, resource_group_name, vm, watcher_rg, watcher_name, location=None):
+
+    if not is_valid_resource_id(vm):
+        vm = resource_id(
+            subscription=get_subscription_id(), resource_group=resource_group_name,
+            namespace='Microsoft.Compute', type='virtualMachines', name=vm)
+
+    return client.get_vm_security_rules(watcher_rg, watcher_name, vm)
+
+
+def create_nw_packet_capture(client, resource_group_name, capture_name, vm,
+                             watcher_rg, watcher_name, location=None,
+                             storage_account=None, storage_path=None, file_path=None,
+                             capture_size=None, capture_limit=None, time_limit=None):
+    from azure.mgmt.network.models import PacketCapture, PacketCaptureStorageLocation
+    storage_settings = PacketCaptureStorageLocation(storage_account, storage_path, file_path)
+    capture_params = PacketCapture(vm, storage_settings, capture_size, capture_limit, time_limit)
+    return client.create(watcher_rg, watcher_name, capture_name, capture_params)
+
+
+def set_nsg_flow_logging(client, watcher_rg, watcher_name, nsg, storage_account=None,
+                         resource_group_name=None, enabled=None, retention=0):
+    try:
+        # update
+        from azure.cli.core.commands import LongRunningOperation
+        config = LongRunningOperation()(client.get_flow_log_status(watcher_rg, watcher_name, nsg))
+        if enabled is not None:
+            config.enabled = enabled
+        if storage_account is not None:
+            config.storage_id = storage_account
+        if retention is not None:
+            from azure.mgmt.network.models import RetentionPolicyParameters
+            config.retention_policy = RetentionPolicyParameters(retention, int(retention) > 0)
+    except CloudError:
+        # create
+        from azure.mgmt.network.models import FlowLogInformation, RetentionPolicyParameters
```

Comment:
Are there any status codes you can check to make sure it was indeed caused by flow log not configured?